### ISSUE TITLE
Full node in monerujo

### ISSUE THREAD

**Issue by Andrewskiz** [m2049r/xmrwallet#538]
I realize this isn't an option for everyone, but now that 128GB, 512GB, and even 1TB MicroSD cards are a thing it might be a nice option to store the blockchain locally and not have to connect to a remote node (even if it's one you're hosting yourself).

**Comment by m2049r** [comment]
there are ideas of making an app to act as a node. i don't think the storage space is the bottleneck but the stress on it - SD cards can't take the stress an SSD can handle. but i am no expert.

**Comment by juanpc2018** [comment]
A Full Node is Not as simply as storing raw data in a big file,
 it needs to be checked/verified & processed "for security", that needs CPU, lots = Battery / Energy.

Having a private Node is one of the best things Monero XMR has, 
much better than Electrum Portable wallets, that need special servers with a private network apart from the real network to communicate with the real network... crazy.
Electrum is c rap...
last month was hacked, and lots of people lost lots of money...

Portable Wallets are Bridges, Not parking lots.
Portable Wallets are Wallets, Not Heavy Duty 24-bolt Diebold vaults. LOL. Jajajajaja 

https://en.wikipedia.org/wiki/Bank_vault
https://en.wikipedia.org/wiki/Safe
https://en.wikipedia.org/wiki/Wallet

some people think on-line e-coin web wallets are like Safe Deposit Boxes, but they are wrong...
https://en.wikipedia.org/wiki/Safe_deposit_box
they get hacked very easy....
MtGox was the 1st, but wont be the last.
NiceHashMiner web wallet, and many other got hacked if you search...

another alternative instead of having a full private Node is to connect to a Pool, most trusted/popular pools have open node.
supportxmr for example...
https://miningpoolstats.stream/monero

but i still think having a private node is better., but requires DDNS and a modem with open UDP UTP ports 18081 the most common, if you want to sync with 4G LTE form the street to your Home Private Desktop/Server.

the Desktop private Node can be set to Fast Sync, "trusting" another Node,
like Monerujo or CAKE portable wallets do.
something like MultiDoge Desktop wallet. https://multidoge.org
awesome...

a Thin Client Desktop wallet with magnetic spinning disc technology its like a Home Safe,
better than a wallet, but Not as good as a Vault. "Full Node"
a Full Node with SSD, is Not as safe/secure as a Thin client or Full Node with Magnetic spinning disc.
SSD is better to sync from 0, "for the 1st time" without bootstrap, but Not for long term storage.
2.5" is Better than 3.5", Big & heavy are more delicate....

microSD cards overheat and shutdown to self protect,
it happens with my Sony x1000v, if i use the scuba case outside water camera overheats.
also formatting wrong the microSD, with a smaller block size to "overclock" kills the card with overheat.
microSD cards need 1Amp at 5vdc to read & write at U3 UHS-II 1000x speeds.
microSD does Not have error correction like CompactFlash....

spinning magnetic technology is far better, lasts a lot more, than SSD storage, tolerates more heat.
SSD starts to loose information if its Not turned-on 1 day every 6 months, 
SSD needs to scrub/refresh/move the information constantly from one cell to other in the background... most have a 1GB cache to move the information... Not to increase Burst speed like magnetic spinning discs.

**Comment by cereal** [comment]
> there are ideas of making an app to act as a node. i don't think the storage space is the bottleneck but the stress on it - SD cards can't take the stress an SSD can handle. but i am no expert.

This is true for writes, but only in rare cases for reads on a cell-level based flash (repeated writes on the same cell will destroy it and reads could bitflip a bit once in ~3000 years IIRC), but those cases are handled by the SD card controller, which exposes a block device to the host.. 

The complete blockchain data itself will only be read in rare cases (when initializing a new wallet, and scanning recently received blocks [in case the wallet was initialized before].. smaller chunks, when mining) - so this shouldn't be a big issue.

IMO you could use ext4 (optimized with data=ordered mode) and that should do the trick.. Cards often break, when you disconnect them during write-operations, but I guess Android unmounts the cards correctly in case of low battery and some of the other edge-case - like disconnecting it manually - can be prevented with a drop of epoxy resin :smiling_imp: [**don't** do that].

**Comment by ghost** [comment]
Perhaps this might be feasible with a highly pruned and sharded node. After all, smartphone storage and computing power will continue to improve. @m2049r mentions that the node could be a separate program, similar to how monerod is a daemon separate from desktop wallets. Excuse my ignorance, but what type of inter-process or inter-app communication does android have that would facilitate this sort of setup? I think a separate node would be the way forwards, because it would allow you to have multiple monero-related apps (xmrig, p2pool, wallet) on the same phone without each maintaining their own copy of the node.

**Comment by nahuhh** [comment]
> because it would allow you to have multiple monero-related apps (xmrig, p2pool, wallet) on the same phone without each maintaining their own copy of the node.

Check out @cryptogrampy pocketnode (pre-beta), termux node (clearnet) or tor termux node (check my latest release 4.2.0 for tor version).

The tor node release has instructions for setting up p2pool

Xmrig is easy too, or can use xmrig for android (app)
